op-node/rollup/derive/l1_block_info.go

```diff
@@ -283,11 +322,19 @@ func L1InfoDeposit(rollupCfg *rollup.Config, sysCfg eth.SystemConfig, seqNumber
 		}
 		l1BlockInfo.BlobBaseFeeScalar = scalars.BlobBaseFeeScalar
 		l1BlockInfo.BaseFeeScalar = scalars.BaseFeeScalar
-		out, err := l1BlockInfo.marshalBinaryEcotone()
-		if err != nil {
-			return nil, fmt.Errorf("failed to marshal Ecotone l1 block info: %w", err)
+		if isInteropButNotFirstBlock(rollupCfg, l2BlockTime) {
```

Comment:
I think we want to move this out above `isEcotoneButNotFirstBlock` and do `if else isEcotoneButNotFirstBlock`

Reply:
agreed, only 1 nit-pick, I find it a bit weird to have `scalars, err := sysCfg.EcotoneScalars()` inside the generic `isInteropButNotFirstBlock` if branch.